node_exporter.go

```diff
@@ -29,6 +30,34 @@ func init() {
 	prometheus.MustRegister(version.NewCollector("node_exporter"))
 }
 
+func handler(w http.ResponseWriter, r *http.Request) {
+	filters := r.URL.Query()["collect[]"]
+	log.Debugln("collect query:", filters)
+
+	nc, err := collector.NewNodeCollector(filters...)
+	if err != nil {
+		log.Warnln("Couldn't create", err)
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte(fmt.Sprintf("Couldn't create %s", err)))
+		return
+	}
+
+	registry := prometheus.NewRegistry()
+	registry.MustRegister(nc)
```

Comment:
This can panic. Let's use the `Register` function instead and return a server error response if it fails.

Reply:
Thanks for pointing this out!
@SuperQ We should [fix this in `mysql_exporter`](https://github.com/prometheus/mysqld_exporter/blob/master/mysqld_exporter.go#L226) as well.